lib/rules/no-restricted-exports.js

```diff
@@ -41,6 +43,19 @@ module.exports = {
                     },
                     additionalProperties: false
                 },
+                {
+                    type: "object",
+                    properties: {
+                        restrictedNamedExportsPattern: {
+                            type: "array",
+                            items: {
+                                type: "string"
+                            },
+                            uniqueItems: true
+                        }
+                    },
+                    additionalProperties: false
+                },
```

Comment:
as discussed in the issue this option would be along with `restrictedNamedExports`, here it is in `anyOf` means only one can be used. can you change this.
```js
{
                    type: "object",
                    properties: {
                        restrictedNamedExports: {
                            type: "array",
                            items: {
                                type: "string"
                            },
                            uniqueItems: true
                        },
                       restrictedNamedExportsPattern: {
                            type: "string"
                       }

                    },
                    additionalProperties: false
                },
```